File: Source/Binary/Creators.Collections/ListConverterCreator.cs
using Mikodev.Binary.Internal.Components;
using System;
using System.Collections.Generic;

namespace Mikodev.Binary.Creators.Collections
{
    internal sealed class ListConverterCreator : IConverterCreator
    {
        private static readonly GenericConverterCreator creator = new GenericConverterCreator(new Dictionary<Type, Type>
        {
            [typeof(List<>)] = typeof(ListConverter<>),
        });

        public Converter GetConverter(IGeneratorContext context, Type type) => creator.GetConverter(context, type);
    }
}

File: code/Binary.FSharp/Creators/ListConverterCreator.fs
namespace Mikodev.Binary.Creators

open Mikodev.Binary
open Mikodev.Binary.Internal

[<CompiledName("FSharpListConverterCreator")>]
type internal ListConverterCreator() =
    interface IConverterCreator with
        member __.GetConverter(context, t) =
            CommonHelper.TryCreateConverter(context, t, typedefof<_ list>, typedefof<ListConverter<_>>)
